pygmt/helpers/decorators.py

```diff
@@ -375,10 +375,16 @@ def new_module(*args, **kwargs):
             for arg, alias in aliases.items():
                 if alias in kwargs and arg in kwargs:
                     raise GMTInvalidInput(
-                        f"Arguments in short-form ({arg}) and long-form ({alias}) can't coexist"
+                        f"Parameters in short-form ({arg}) and long-form ({alias}) can't coexist."
                     )
                 if alias in kwargs:
                     kwargs[arg] = kwargs.pop(alias)
+                elif arg in kwargs:
+                    msg = (
+                        f"Short-form parameter ({arg}) is not recommended. "
+                        f"Use long-form parameter '{alias}' instead."
+                    )
+                    warnings.warn(msg, category=FutureWarning, stacklevel=2)
```

Comment:
Using `FutureWarning` implies a deprecation at some future date. Do you want to either:

1. set a specific PyGMT version (e.g. v0.5.0) to deprecate this 'feature'
2. Raise a [SyntaxWarning](https://docs.python.org/3/library/exceptions.html#SyntaxWarning) instead
3. Raise a [SyntaxError](https://docs.python.org/3/library/exceptions.html#SyntaxError) or GMTInvalidInput error?

Reply:
> Using `FutureWarning` implies a deprecation at some future date. Do you want to either:
> 
> 1. set a specific PyGMT version (e.g. v0.5.0) to deprecate this 'feature'

As mentioned in https://github.com/GenericMappingTools/pygmt/issues/1203#issuecomment-845253702, I don't think we will fully deprecate this feature before PyGMT v1.0.0.

> 2. Raise a [SyntaxWarning](https://docs.python.org/3/library/exceptions.html#SyntaxWarning) instead

Sounds good to me.

> 3. Raise a [SyntaxError](https://docs.python.org/3/library/exceptions.html#SyntaxError) or GMTInvalidInput error?

I'm afraid it's a little too aggressive. BTW, the "single-letter" parameter is also used a lot in the PyGMT tests. Raising any errors will bring us troubles.

Reply:
Changed to `SyntaxWarning` in commit 23e9dd67a999cb0c984dea13fb993434f4593944.